### PR TITLE
[Fix #313] `pf` doesn't understand %&

### LINKED ISSUE #313: Promote-function doesn't understand %&
With a function literal like `#(apply str %&)`, the command will prompt for a name to use for `%`, and the `&` character will not be deleted.

### PR DESCRIPTION
**Pull request by expez**
No tests are added because of a long standing bug in the test framework
where use of `%s` triggers a 'wrong number of args passed to format'
error.

### CODE REVIEW

**Comment by expez** [comment]
The tests pass locally, not sure what the weird stuff on travis is about.


````
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
Warning (emacs): Unrecognized key: _r_
```

Times a million :(

**Comment by Malabarba** [comment]
> where use of %s triggers a 'wrong number of args passed to format'

Sorry if this was already suggested, but is it possible to use `%%s` instead?

File: clj-refactor.el [hunk 1]
```diff
@@ -2611,10 +2611,12 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-update-project-d
   (cljr--goto-fn-definition)
   (let ((fn-start (point))
         var replacement)
-    (while (re-search-forward "%[1-9]?" (cljr--point-after 'paredit-forward) t)
+    (while (re-search-forward "%[1-9]?&?" (cljr--point-after 'paredit-forward) t)
```
Comment (Malabarba) [inline clj-refactor.el]: Or `"%[1-9&]?"`?

**Comment by expez** [comment]
> Sorry if this was already suggested, but is it possible to use `%%s` instead?

Unfortunately not :(

**Comment by benedekfazekas** [comment]
I guess you mean this [bug](https://github.com/ecukes/ecukes/issues/58) there might be a workaround altho ugly. can you pls add the test to the commit or just post a patch file so I can play around a bit with it.

travis failure: weird. everything passes locally for me too.

**Comment by expez** [comment]
There is no workaround, short of coupling our tests to the number of calls to `format` used in the library.  I'm not going to count, nor maintain such a fragile test.

Looks like we got a bad container or something, the travis tests are passing again on this branch.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 ### Bugs fixed
 
+- [#313](https://github.com/clojure-emacs/clj-refactor.el/issues/313) `*data-readers*` teach `pf` about function literals using `%&`.
 - [#320](https://github.com/clojure-emacs/clj-refactor.el/issues/320) `*data-readers*` ignored when searching for macros.
```
Comment (expez) [inline CHANGELOG.md]: Lol

 that copy paste error propagated through all my commits

**Comment by benedekfazekas** [comment]
> There is no workaround, short of coupling our tests to the number of calls to format used in the external test library. I'm not going to count, nor maintain such a fragile test.

fair enough

**Comment by benedekfazekas** [comment]
> There is no workaround, short of coupling our tests to the number of calls to format used in the external test library. I'm not going to count, nor maintain such a fragile test.

fair enough

**Pull request merged**